src/components/ecl-button/src/styles/ecl-button-ec.scss

```diff
@@ -1,24 +1,25 @@
-@use '@ecl/theme-ec';
-@use '@ecl/vanilla-component-icon/icon-ec';
-@use '@ecl/vanilla-component-link/link-ec';
-@use '@ecl/vanilla-component-button/button-ec';
+@use '@ecl/theme-ec/theme' as theme;
+@use '@ecl/theme-ec/variables' as var;
+@use '@ecl/grid/grid' with (
+  $theme: theme.$theme
+);
+@use '@ecl/button/button' with (
+  $theme: theme.$theme,
+  $button: var.$button
+);
 
 :host {
   display: contents;
 }
 
-.ecl-button__icon {
-  vertical-align: middle;
+.ecl-button--has-label ::slotted([slot='icon-before']) {
```

Comment:
am i wrong ot this is adding a double spacing to the icon..? Due to the fact that we have this ecl-button__icon-container class everywhere we have an icon and in ECL that gets the same spacing as the icon itself, but in ECL this element is only present when there is an indicator, so if you refactor the markup then probably this will be needed.